binary_tree.rb

```diff
@@ -0,0 +1,65 @@
+require 'bundler'
+
+Bundler.require(:default)
+
+class BinaryTree
+  attr_reader :left
+  attr_reader :right
+  attr_reader :value
+
+  def initialize(left, right, value)
+    @left = left
+    @right = right
+    @value = value.to_s
+  end
+
+  def self.create(array)
+
+    array.sort!
+    array_length = array.length
+
+    if array_length.odd?
```

Comment:
Drobna uwaga - tak na prawdę algorytm mógłby używać po prostu `root_index = (array_length/2)` niezależnie od parzystości, o ile tylko upewnimy się, że nie próbujemy potem wywołać metody `create` z argumentem `nil` (np. przypisując pustą tablicę zamiast nil `array_right = array[(root_index+1...array_length)] || []`. To podejście jest w mojej ocenie nieco bardziej czytelne, natomiast to trochę kwestia preferencji.